include/ddc/kernels/splines/splines_linear_problem_maker.hpp

```diff
@@ -17,6 +18,42 @@ namespace ddc::detail {
 class SplinesLinearProblemMaker
 {
 public:
+    /**
+     * @brief Construct a dense matrix
+     *
+     * @tparam the Kokkos::ExecutionSpace on which matrix-related operation will be performed.
+     * @param n The size of one of the dimensions of the square matrix.
+     *
+     * @return The SplinesLinearProblem instance.
+     */
+    template <typename ExecSpace>
+    static std::unique_ptr<SplinesLinearProblem<ExecSpace>> make_new_dense(int const n)
+    {
+        return std::make_unique<SplinesLinearProblemDense<ExecSpace>>(n);
+    }
+
+    /**
+     * @brief Construct a band matrix
+     *
+     * @tparam the Kokkos::ExecutionSpace on which matrix-related operation will be performed.
+     * @param n The size of one of the dimensions of the square matrix.
+     * @param kl The number of subdiagonals.
+     * @param ku The number of superdiagonals.
+     * @param pds A boolean indicating if the matrix is positive-definite symetric or not.
+     *
+     * @return The SplinesLinearProblem instance.
+     */
+    template <typename ExecSpace>
+    static std::unique_ptr<SplinesLinearProblem<ExecSpace>> make_new_band(
+            int const n,
+            [[maybe_unused]] int const kl,
+            [[maybe_unused]] int const ku,
+            [[maybe_unused]] bool const pds)
+    {
+        // Placeholder while SplinesLinearProblemBand is not implemented
+        return std::make_unique<SplinesLinearProblemDense<ExecSpace>>(n);
+    }
```

Comment:
Why do we need to introduce this function ?

Reply:
You mean why now ? Because matrix tests use it

Reply:
Ok, I think we will need to change the way we do the testing. We need to ensure that all matrices are tested. Using the `make_*` functions, it is not clear in the end if from the runtime parameters we cover all matrices.

Reply:
Add explicit simple tests which bypass SplinesLinearProblemMaker (by directly calling the constructors) would be enough ? And keep existing tests which look usefull to explore a large panel of possible problems.

Reply:
Yes exactly

Reply:
done in cb151bc